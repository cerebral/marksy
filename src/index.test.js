import React from 'react';
import renderer from 'react-test-renderer';
import marksy from 'marksy'

class TestComponent extends React.Component {
  render () {
    return (
      <div>
        {this.props.children}
      </div>
    )
  }
} 

it('should be able to compile text', () => {
  const compile = marksy();
  const compiled = compile(`hello`);
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should be able to compile strong text', () => {
  const compile = marksy();
  const compiled = compile(`hello **there**`);
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should be able to compile strong text', () => {
  const compile = marksy();
  const compiled = compile(`hello **there**`);
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should be able to compile italic text', () => {
  const compile = marksy();
  const compiled = compile(`hello *there*`);
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should be able to compile links', () => {
  const compile = marksy();
  const compiled = compile(`[my link](http://example.com)`);
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should be able to compile headers', () => {
  const compile = marksy();
  const compiled = compile(`
# header1
## header2
### header3
#### header4
  `)
  const tree = renderer.create(
    <TestComponent>{compiled.tree}</TestComponent>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});