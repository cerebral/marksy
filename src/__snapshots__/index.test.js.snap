// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to compile headers 1`] = `
<div>
  <h1
    id="header1"
  >
    header1
  </h1>
  <h2
    id="header2"
  >
    header2
  </h2>
  <h3
    id="header3"
  >
    header3
  </h3>
  <h4
    id="header4"
  >
    header4
  </h4>
</div>
`;

exports[`should be able to compile italic text 1`] = `
<div>
  <p>
    hello 
    <em>
      there
    </em>
  </p>
</div>
`;

exports[`should be able to compile links 1`] = `
<div>
  <p>
    <a
      href="http://example.com"
      target="new"
      title={null}
    >
      my link
    </a>
  </p>
</div>
`;

exports[`should be able to compile strong text 1`] = `
<div>
  <p>
    hello 
    <strong>
      there
    </strong>
  </p>
</div>
`;

exports[`should be able to compile strong text 2`] = `
<div>
  <p>
    hello 
    <strong>
      there
    </strong>
  </p>
</div>
`;

exports[`should be able to compile text 1`] = `
<div>
  <p>
    hello
  </p>
</div>
`;
